### ISSUE TITLE
Sorting of results when using MemoryErrorLog is wrong

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Benknightdark/elmah#289]
```
What steps will reproduce the problem?
1. Have enough errors so that you have several pages
2. Make sure the errors are spaced in time sufficiently

Expected to see the most recent issues on the first page, instead it shows the 
oldest issues first, and the newest issues in the last page.


```

Original issue reported on code.google.com by `deathrav...@gmail.com` on 16 Apr 2012 at 8:17

**Comment by GoogleCodeExporter** [comment]
```
After seeing its still not fixed, i revisited the issue. Its still a problem.

We are using the 64x version. The problem only manifests itself when we have 
more then 100 errors (eg 150). And then it only happens when limiting the page 
to 10 - 50 items. When we set the pagesize to 100 the problem goes away, and 
the sorting is correct.

I have taken a look at the code in the InMemoryLog, but apparently its not a 
straight forward bug. I cant really explain why the sorting would be wrong with 
pagesizes smaller then 100.
```

Original comment by `deathrav...@gmail.com` on 6 Dec 2012 at 9:57

**Comment by GoogleCodeExporter** [comment]
```
Which version are you using?
```

Original comment by `azizatif` on 6 Dec 2012 at 9:06

**Comment by GoogleCodeExporter** [comment]
Original comment by `azizatif` on 6 Dec 2012 at 9:07
- Changed title: **Sorting of results when using MemoryErrorLog is wrong**

**Comment by GoogleCodeExporter** [comment]
```
The latest
```

Original comment by `deathrav...@gmail.com` on 6 Dec 2012 at 9:24

**Comment by GoogleCodeExporter** [comment]
```
Attached is an F# script, including output, that reproduces this issue and 
there indeed is a bug in the MemoryErrorLog implementation of GetErrors.

Attached files also posted online as a gist at:
https://gist.github.com/4261024

```

Original comment by `azizatif` on 11 Dec 2012 at 6:59
- Changed state: **Accepted**
- Added labels: **Component-Persistence**, **Milestone-Release2.0**, **Priority-High**
- Removed labels: **Priority-Medium**

Attachments:
- [issue289.fsx](https://storage.googleapis.com/google-code-attachments/elmah/issue-289/comment-5/issue289.fsx)
- [issue289.txt](https://storage.googleapis.com/google-code-attachments/elmah/issue-289/comment-5/issue289.txt)

**Comment by GoogleCodeExporter** [comment]
```
Issue 341 has been merged into this issue.
```

Original comment by `azizatif` on 21 Aug 2013 at 2:13

**Comment by GoogleCodeExporter** [comment]
```
https://code.google.com/r/thomasardal-issue289/
```

Original comment by `azizatif` on 22 Aug 2013 at 6:31
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
@thomasardal Fix looking good so far. I adjusted the F# script based on ELMAH 
API changes and attached it here together with the output
```

Original comment by `azizatif` on 22 Aug 2013 at 7:06

Attachments:
- [issue289.fsx](https://storage.googleapis.com/google-code-attachments/elmah/issue-289/comment-8/issue289.fsx)
- [issue289.txt](https://storage.googleapis.com/google-code-attachments/elmah/issue-289/comment-8/issue289.txt)

**Comment by GoogleCodeExporter** [comment]
```
When the patch/change is integrated in the current trunk. I can test it as well 
(probably somewhere next week).

Great work.

Im Curious as to what the actual problem was, but ill wait for the revision 
diff.
```

Original comment by `deathrav...@gmail.com` on 22 Aug 2013 at 9:52

**Comment by GoogleCodeExporter** [comment]
```
Great! Would be nice with some tests before this one is released.
```

Original comment by `thomasar...@gmail.com` on 22 Aug 2013 at 10:09

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision ac5eb0db5619.
```

Original comment by `thomasar...@gmail.com` on 22 Aug 2013 at 1:38
- Changed state: **Fixed**
